### DISCUSSION TITLE
GCAM6.0: New Fuel addition

### DISCUSSION THREAD

**Discussion by chacha0906** [JGCRI/gcam-core#325] [Modifying GCAM]
How do I add a new fuel, say "X," in GCAM 6.0 or any other version where it could be easier? I know that fuel "X" is generated from feedstocks hydrogen and carbon. I just wanted to know how to add this new fuel. I have fuel "X" data on its emissions, feedstocks, economic values, etc.

**Comment by pkyle** [comment]
If you search the string "BTL with hydrogen" that should pull up everything you need. This is a "new" refining technology that doesn't play in the historical years. Just add a row (or several) with the technology's characteristics to each of these files.
```
> grep -rl "BTL with hydrogen" input/gcamdata/inst/extdata/*
input/gcamdata/inst/extdata/energy/A22.globaltech_shrwt.csv
input/gcamdata/inst/extdata/energy/A22.globaltech_input_driver.csv
input/gcamdata/inst/extdata/energy/A22.globaltech_cost.csv
input/gcamdata/inst/extdata/energy/A22.globaltech_cost_low.csv
input/gcamdata/inst/extdata/energy/A22.globaltech_keyword.csv
input/gcamdata/inst/extdata/energy/A22.globaltech_coef.csv
input/gcamdata/inst/extdata/energy/A22.globaltech_retirement.csv
```

**Reply by chacha0906** [reply]
Thanks for your considerate reply Kyle, i have gone through your suggested files, the problem is "BTL with hydrogen" is being added as a new technology. I wanted to create a new fuel. Though I get your point as "BTL with hydrogen" is getting "H2 industrial" as the minicam energy input. My fuel X would contain hydrogen and CO2 both as inputs and would have its carbon emissions associated which is needed to be added as well. Should I add a new technology for my fuel?

**Reply by pkyle** [reply]
In GCAM "refining" or "refined liquids" is understood to be include a number of different fuels that aren't necessarily substitutable in the end-use applications, but whose basic characteristics are similar enough that they can reasonably share a single market price and a single carbon content (19.6 kgC/GJ). E.g. gasoline, diesel, jet fuel, many others. If you want to include a new e-fuel commodity that has its own market price, and its own supply pathway to end-use or T&D technologies that take it specifically as an input, then the method I gave you above won't work, and I don't have any "off the shelf" examples to share. Still for what you're doing I think a technology within refining would work. AFAIK you can't assign an input of CO2, but the accounting of the carbon would be correct; there would be net-zero CO2 emissions from the fuel's production and use.
